src/proof/dot/dot_printer.h

```diff
@@ -42,24 +84,33 @@ class DotPrinter
  private:
   /**
    * Print the nodes of the proof in the format:
-   * $NODE_ID [ label = "{$CONCLUSION|$RULE_NAME($RULE_ARGUMENTS)}",
-   * $COLORS_AND_CLASSES_RELATED_TO_THE_RULE ]; and then for each child of the
+   * $NODE_ID [ label = "{$CONCLUSION|$RULE_NAME($RULE_ARGUMENTS)}", comment =
+   * "{\"subProofQty\":$SUB_PROOF_QUANTITY}" ]; and then for each child of the
    * node $CHILD_ID -> $NODE_ID; and then recursively calls the function with
    * the child as argument.
    * @param out the output stream
    * @param pn the proof node to print
-   * @param pfLet the map of the hashs of proof nodes already printed to their ids
-   * @param scopeCounter counter of how many SCOPE were already depth-first
-   * traversed in the proof up to this point
-   * @param inPropositionalView flag used to mark the proof node being traversed
-   * was generated by the SAT solver and thus belong to the propositional view
+   * @param pfLet the map of the hashs of proof nodes already printed to their
+   * ids
+   * @param parentType the type of the parent node
    * @return the id of the proof node printed
    */
   uint64_t printInternal(std::ostream& out,
                          const ProofNode* pn,
                          std::map<size_t, uint64_t>& pfLet,
-                         uint64_t scopeCounter,
-                         bool inPropositionalView);
+                         ProofNodeClusterType parentType);
+
+  /**
+   * Print the nodes of the proof in the format:
+   * $NODE_ID [ label = "{$CONCLUSION|$RULE_NAME($RULE_ARGUMENTS)}", comment =
+   * "{\"subProofQty\":$SUB_PROOF_QUANTITY}" ];
+   * @param out the output stream
+   * @param pn the proof node to print
+   * @param currentRuleID the current rule ID
+   */
+  inline void printProofNodeInfo(std::ostream& out,
+                                 const ProofNode* pn,
+                                 uint64_t currentRuleID);
```

Comment:
You can remove this parameter since you are always printing the information of the current rule id.